src/main/java/software/amazon/encryption/s3/S3AsyncEncryptionClient.java

```diff
@@ -719,6 +741,34 @@ public Builder useArnRegion(Boolean useArnRegion) {
             return this;
         }
 
+        /**
+         * Multipart via the wrapped client is currently NOT supported by the S3 Encryption Client.
+         * Use the {@link this.enableMultipartPutObject()} option instead for high-level multipart uploads.
+         * Multipart downloads are currently NOT supported.
+         * @param enabled MUST be false
+         * @return
+         */
+        @Override
+        public Builder multipartEnabled(Boolean enabled) {
+            if (enabled) {
+                throw new UnsupportedOperationException("The S3 Encryption Client does not support wrapped clients with automatic multipart enabled.");
+            }
+            _multipartEnabled = enabled;
+            return this;
+        }
+
+        @Override
+        public Builder multipartConfiguration(MultipartConfiguration multipartConfiguration) {
+            _multipartConfiguration = multipartConfiguration;
+            return this;
+        }
```

Comment:
I am confused.
Do we support `software.amazon.awssdk.services.s3.multipart.MultipartConfiguration`?
I would assume that we do NOT.

Just like we do not support `multipareEnabled`.

Here is the JavaDoc for `MultipartConfiguration`:
```
Class that hold configuration properties related to multipart operation for a S3AsyncClient. Passing this class to the S3AsyncClientBuilder. multipartConfiguration(MultipartConfiguration) will enable automatic conversion of S3AsyncClient. getObject(GetObjectRequest, AsyncResponseTransformer), S3AsyncClient. putObject(PutObjectRequest, AsyncRequestBody) and S3AsyncClient. copyObject(CopyObjectRequest) to their respective multipart operation.
Note that multipart download fetch individual part of the object using part number, this means it will only download multiple parts if the object itself was uploaded as a multipart object
```

Furthermore, I found this class in S3 Client:
`software.amazon.awssdk.services.s3.internal.client.S3AsyncClientDecorator`.

The relevant code is:
```
        decorators.add(ConditionalDecorator.create(
            isMultipartEnable(clientContextParams),
            client -> {
                MultipartConfiguration multipartConfiguration = clientContextParams.get(MULTIPART_CONFIGURATION_KEY);
                return MultipartS3AsyncClient.create(client, multipartConfiguration);
            }));
```

So either of these builder options should trigger an error.